osi_object.proto

```diff
@@ -745,66 +748,133 @@ message MovingObject
+            // Vehicle is a (heavy) truck.
+            //
+            // Definition: A rigid-body truck that cannot pull a semi-trailer.
+            //
+            // This corresponds to "truck" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "truck" in OpenScenario 2.x.
             //
             TYPE_HEAVY_TRUCK = 7;
 
-            // Vehicle is a truck with semitrailer.
+            // Vehicle is a tractor unit capable of pulling a semi-trailer.
+            //
+            // Definition: The cab for an articulated vehicle.
+            //
+            // This has no equivalent in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "truck" in OpenScenario 2.x.
+            //
+            TYPE_SEMITRACTOR = 16;
+
+            // This vehicle is a trailer unit for an articulated vehicle.
+            //
+            // Definition: A trailer that can be pulled by a semi-tractor.
+            // There is no precise definition of this vs a typical trailer, although
+            // typically it is used for trailers with a non-trivial bounding box
+            // overlap with the tractor unit.
+            //
+            // This corresponds to "semitrailer" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "trailer" in OpenScenario 2.x.
             //
             TYPE_SEMITRAILER = 8;
 
             // Vehicle is a trailer (possibly attached to another vehicle).
             //
+            // Definition: A trailer that isn't pulled by a semi-tractor.
+            //
+            // This corresponds to "trailer" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "trailer" in OpenScenario 2.x.
+            //
             TYPE_TRAILER = 9;
 
             // Vehicle is a motorbike or moped.
             //
+            // This corresponds to "motorbike" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "VRU_vehicle" in OpenScenario 2.x.
+            //
             TYPE_MOTORBIKE = 10;
 
             // Vehicle is a bicycle (without motor and specific lights).
             //
+            // This corresponds to "bicycle" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "bus" in OpenScenario 2.x.
+            //
             TYPE_BICYCLE = 11;
 
             // Vehicle is a bus.
             //
+            // This corresponds to "bus" in both OpenScenario 1.x and 2.x.
+            //
+            // This is one of the OSI types which map from "bus" in OpenScenario 2.x.
+            //
             TYPE_BUS = 12;
 
             // Vehicle is a tram.
             //
+            // This corresponds to "tram" in OpenScenario 1.x.
+            //
+            // This is one of the OSI types which map from "bus" in OpenScenario 2.x.
```

Comment:
I am not sure OpenSCENARIO 2.0 intends to describe a tram with bus. Did you talk with the Domain Model people there? While things are still under development there, they currently have an `intended_infrastructure` enum with the value `tram `while (you are right here) no value in `vehicle_category` for tram. Unless clarified I think I would delete this line here.

Reply:
Yes, I must confess this was a guess :)

The definition in OSC 2.0 is to cover UN categories M2 and M3.  They are:

- 2.2.2. "Category M2": Vehicles used for the carriage of passengers, comprising more than eight seats in addition to the driver's seat, and having a maximum mass not exceeding 5 tonnes.
- 2.2.3. "Category M3": Vehicles used for the carriage of passengers, comprising more than eight seats in addition to the driver's seat, and having a maximum mass exceeding 5 tonnes.

So, trams and trains match both of those, but I guess it depends what you define "vehicle" to be.

It would be great to chat to the OSC team to work out their intention.

Reply:
They are currently very busy, but I can post it as a question in the review process which is currently ongoing over there... if that' fine for you.

Reply:
Yes, perfect, thanks.

I don't think there is a rush here, I mainly put this in to highlight that we should be aiming to provide this kind of detail wherever we can.

Reply:
Thanks for raising!

I have removed any reference to 2.x from both tram and train while this discussion is ongoing.